cunumeric/module.py

```diff
@@ -6242,6 +6242,142 @@ def nansum(
 # Arithmetic operations
 
 
+@add_boilerplate("a")
+def diff(
+    a: ndarray,
+    n: int = 1,
+    axis: int = -1,
+    prepend: Any = None,
+    append: Any = None,
+) -> ndarray:
```

Comment:
With this you can skip the `asanyarray` below, you'll already have an `ndarray` if it was not `None`.

```suggestion
@add_boilerplate("a", "prepend", "append")
def diff(
    a: ndarray,
    n: int = 1,
    axis: int = -1,
    prepend: ndarray | None = None,
    append: ndarray | None = None,
) -> ndarray:
```